engine-tests/src/tests/eth_connector.rs

```diff
@@ -457,6 +457,100 @@ fn test_ft_transfer_call_eth() {
     assert_eq!(balance, transfer_amount);
 }
 
+#[test]
+fn test_ft_transfer_call_without_message() {
+    let (master_account, contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_eth_to_near(&contract, CONTRACT_ACC);
+
+    let balance = get_eth_on_near_balance(&master_account, DEPOSITED_RECIPIENT, CONTRACT_ACC);
+    assert_eq!(balance, DEPOSITED_AMOUNT - DEPOSITED_FEE);
+
+    let balance = get_eth_on_near_balance(&master_account, CONTRACT_ACC, CONTRACT_ACC);
+    assert_eq!(balance, DEPOSITED_FEE);
+
+    let res = contract.call(
+        CONTRACT_ACC.parse().unwrap(),
+        "register_relayer",
+        &RegisterRelayerCallArgs {
+            address: validate_eth_address(CUSTODIAN_ADDRESS),
+        }
+        .try_to_vec()
+        .unwrap(),
+        DEFAULT_GAS,
+        0,
+    );
+    res.assert_success();
+
+    let transfer_amount = 50;
+    // Send to Aurora contract with wrong message should failed
+    let res = contract.call(
+        CONTRACT_ACC.parse().unwrap(),
+        "ft_transfer_call",
+        json!({
+            "receiver_id": CONTRACT_ACC,
+            "amount": transfer_amount.to_string(),
+            "msg": "",
+        })
+        .to_string()
+        .as_bytes(),
+        DEFAULT_GAS,
+        1,
+    );
+    match res.outcome().status {
+        ExecutionStatus::Failure(_) => {}
+        _ => panic!("should panice"),
+    }
+
+    // Sending to excternal receiver with empty message should be success
+    let transfer_amount = 22;
+    let res = contract.call(
+        CONTRACT_ACC.parse().unwrap(),
+        "ft_transfer_call",
+        json!({
+            "receiver_id": "some-test-acc",
```

Comment:
It's ok to test the call to `some-test-acc`, but as it obviously doesn't have a `ft_on_transfer` method, it makes sense to also test against some test contract that indeed has `ft_on_transfer` method (could be empty, or just an accounting one).

Reply:
From the [NEP-141 standard](https://nomicon.io/Standards/FungibleToken/Core.html):

```
// * The receiving contract must implement `ft_on_transfer` according to the
//   standard. If it does not, FT contract's `ft_resolve_transfer` MUST deal
//   with the resulting failed cross-contract call and roll back the transfer.
```

So if `some-test-acc` does not implement `ft_on_transfer` then the transfer should be undone, no happen twice! This looks like a bug in our FT implementation.

Reply:
Yes, that's what I'm talking about and why it's important to have both cases covered in tests. Perhaps on https://github.com/aurora-is-near/aurora-engine/blob/c5454c15c12f83034f82acd7849f91cc858f58f5/engine/src/fungible_token.rs#L274 we also need to check for `promise_results_count`?

Reply:
> makes sense to also test against some test contract that indeed has `ft_on_transfer` method

New test added in https://github.com/aurora-is-near/aurora-engine/pull/326/commits/28953c6f4b4bf43be5f5df56ba7bc778ec6877d4

> we also need to check for `promise_results_count`?

This [check is done `EthConnectorContract::ft_resolve_transfer`](https://github.com/aurora-is-near/aurora-engine/blob/c5454c15c12f83034f82acd7849f91cc858f58f5/engine/src/connector.rs#L471) so it does not need to repeated. The code could be refactored to make it clear that check is already done, but in the spirit of doing a minimal change for the issues at hand I think we should do it in a future PR.